### PR TITLE
View Command Implementation and model interaction

### LINKED ISSUE #104: Create/Implement View Command

### LINKED ISSUE #105: Add test for View Command (Week8/9)

### LINKED ISSUE #110: Switch to TBM for first MVP
NTS: PR straight to master for this.

### PR DESCRIPTION
**Pull request by LeeEnHao**
- [x] Fixes #104

- [x] Fixes #105 

- [x] Fixes #110

### CODE REVIEW

File: src/main/java/seedu/address/logic/commands/ClientViewCommand.java [hunk 1]
```diff
@@ -0,0 +1,61 @@
...
+import seedu.address.commons.core.Messages;
+import seedu.address.commons.core.index.Index;
+import seedu.address.logic.commands.exceptions.CommandException;
+import seedu.address.model.Model;
+import seedu.address.model.client.Client;
+
+public class ClientViewCommand extends Command {
+
+    /*Client command in the form client view index, where index points to the index of teh client at current location
+    of the rendered */
```
Comment (tankangliang) [inline src/main/java/seedu/address/logic/commands/ClientViewCommand.java]:
```suggestion
    /** Client command in the form client view index, where index points to the index of the client at current location
    of the rendered */
```

**Review by LeeEnHao — COMMENTED** [review]
S 2a2e7a5... d632de7

**Comment by LeeEnHao** [comment]
Adding of the view box to the GUI will be done in separate branch in case GUI breaks.

File: src/main/java/seedu/address/logic/Logic.java [hunk 2]
```diff
@@ -30,6 +31,12 @@
      */
     ReadOnlyAddressBook getAddressBook();
 
+    /**
+     * Returns the content of the widget view box.
+     * @return
```
Comment (tankangliang) [inline src/main/java/seedu/address/logic/Logic.java]: Should fill in the javadoc here

File: src/main/java/seedu/address/logic/commands/ClientViewCommand.java [hunk 3]
```diff
@@ -0,0 +1,62 @@
...
+import seedu.address.logic.commands.exceptions.CommandException;
+import seedu.address.model.Model;
+import seedu.address.model.client.Client;
+
+/**
+ * Views a client identified by its displayed index. Displayed on the widget view box.
+ */
+public class ClientViewCommand extends Command {
+
+    /*Client command in the form client view index, where index points to the index of the client at current location
```
Comment (tankangliang) [inline src/main/java/seedu/address/logic/commands/ClientViewCommand.java]:
```suggestion
    /** Client command in the form client view index, where index points to the index of the client at current location
```
To be more inline with the coding standards 
image

File: src/main/java/seedu/address/logic/parser/ViewCommandParser.java [hunk 4]
```diff
@@ -0,0 +1,28 @@
...
+import static seedu.address.commons.core.Messages.MESSAGE_INVALID_COMMAND_FORMAT;
+
+import seedu.address.commons.core.index.Index;
+import seedu.address.logic.commands.ClientViewCommand;
+import seedu.address.logic.parser.exceptions.ParseException;
+
+/**
+ * Parses input arguments and creates a new ClientViewCommand object
+ */
+public class ViewCommandParser implements Parser<ClientViewCommand> {
```
Comment (tankangliang) [inline src/main/java/seedu/address/logic/parser/ViewCommandParser.java]: Could you change the class and file name to `ClientViewCommandParser`? I will be updating the ones for Find and Delete in a future PR

File: src/main/java/seedu/address/model/widget/WidgetModel.java [hunk 5]
```diff
@@ -0,0 +1,34 @@
...
+ *
+ * v1.2 - display contents are limited to objects of Client type.
+ * v1.3 - ???
+ */
+public interface WidgetModel {
+
+    /**
+     * Static factory.
+     *
+     * @return
```
Comment (tankangliang) [inline src/main/java/seedu/address/model/widget/WidgetModel.java]: Missing javadoc for return here too

**Review by tankangliang — CHANGES_REQUESTED** [review]
Tested `client view` with negative index, valid index and non-numbers and seems to be working fine. Thanks for adding the icon and title for TBM. Haven't been able to look at widget component but it doesn't seem to be affecting any functionality. Other than minor javadoc comments, looks good to merge.

File: src/main/java/seedu/address/logic/commands/ClientViewCommand.java [hunk 6]
```diff
@@ -0,0 +1,62 @@
...
+import seedu.address.logic.commands.exceptions.CommandException;
+import seedu.address.model.Model;
+import seedu.address.model.client.Client;
+
+/**
+ * Views a client identified by its displayed index. Displayed on the widget view box.
+ */
+public class ClientViewCommand extends Command {
+
+    /**Client command in the form client view index, where index points to the index of the client at current location
```
Comment (tankangliang) [inline src/main/java/seedu/address/logic/commands/ClientViewCommand.java]:
```suggestion
    /** Client command in the form client view index, where index points to the index of the client at current location
```
Pls have the space there :(

Comment (LeeEnHao) [reply src/main/java/seedu/address/logic/commands/ClientViewCommand.java]: Oh lol ps

**Review by tankangliang — APPROVED** [review]

**Comment by LeeEnHao** [comment]
Integrity of Country behaviour seems intact. I'll merge now?

File: src/main/java/seedu/address/model/Model.java [hunk 7]
```diff
@@ -70,7 +71,22 @@
     void deleteClient(Client target);
 
     /**
-     * Adds the given client. {@code client} must not already exist in the address book.
+     * Sets the widget box contents to the given content.
+     */
+    void setWidgetContent(Object content);
```
Comment (raysonkoh) [inline src/main/java/seedu/address/model/Model.java:76]: Should `setWidgetContent` be in `Model`? Seems like this is mixing UI-related stuff with model-related stuff.

**Review by raysonkoh — APPROVED** [review]
Generally looks ok. Some minor concerns with `setWidgetContent` being in `Model` but if it works for now then this can be a future discussion/improvement.

**Pull request merged**